src/core/src/OmniTileset.cpp

```diff
@@ -504,6 +505,30 @@ void OmniTileset::reload() {
             }
         }
     }
+
+    const auto& allOmniPolygonRasterOverlays = _pContext->getAssetRegistry().getPolygonRasterOverlays();
+    const auto& boundNativeRasterOverlays =
+        _pTileset->getOverlays().getOverlays(); // get RasterOverlays from RasterOverlayCollection
+    for (const auto& pOmniPolygonRasterOverlay : allOmniPolygonRasterOverlays) {
+        const auto& pNativeOverlay = pOmniPolygonRasterOverlay->getRasterOverlay();
+
+        auto matchIterator = std::find_if(
+            boundNativeRasterOverlays.begin(),
+            boundNativeRasterOverlays.end(),
+            [&pNativeOverlay](const auto& possibleMatch) { return possibleMatch.get() == pNativeOverlay; });
+
+        if (matchIterator == boundNativeRasterOverlays.end()) {
+            continue;
+        }
+
+        const auto exclude = pOmniPolygonRasterOverlay->getExcludeSelectedTiles();
+        if (exclude) {
+            const auto excluder = pOmniPolygonRasterOverlay->getExcluder();
+            if (excluder) {
+                _pTileset->getOptions().excluders.push_back(excluder);
+            }
+        }
+    }
```

Comment:
This could be simplified to something like:

```c++
    const auto rasterOverlayPaths = getRasterOverlayPaths();

    for (const auto& rasterOverlayPath : rasterOverlayPaths) {
        const auto pPolygonRasterOverlay = _pContext->getAssetRegistry().getPolygonRasterOverlay(rasterOverlayPath);
        if (pPolygonRasterOverlay) {
            const auto pExcluder = pPolygonRasterOverlay->getExcluder();
            if (pExcluder) {
                options.excluders.push_back(pExcluder);
            }
        }
    }
```

and moved above so that it's not modifying options after the tileset has been constructed.

Reply:
Yeah, that's a lot simpler than querying the `RasterOverlayCollection`.